File: app/src/main/java/net/squanchy/tweets/view/TweetViewHolder.java
package net.squanchy.tweets.view;

import android.support.v7.widget.RecyclerView.ViewHolder;
import android.view.View;

import net.squanchy.tweets.domain.view.TweetViewModel;

class TweetViewHolder extends ViewHolder {

    TweetViewHolder(View itemView) {
        super(itemView);
    }

    void updateWith(TweetViewModel tweet) {
        ((TweetItemView) itemView).updateWith(tweet);
    }
}

File: app/src/main/java/net/squanchy/tweets/view/TweetViewHolder.kt
package net.squanchy.tweets.view

import android.view.View
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import net.squanchy.tweets.domain.TweetLinkInfo
import net.squanchy.tweets.domain.view.TweetViewModel

class TweetViewHolder(itemView: View) : ViewHolder(itemView) {

    fun updateWith(tweet: TweetViewModel, listener: (TweetLinkInfo) -> Unit) {
        val tweetView = itemView as TweetItemView
        tweetView.updateWith(tweet, listener)
    }
}
